### PR TITLE
[Quant][PT2E]Fix pt2e quantization maxpool input observer issue

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98826
* #98730
* __->__ #100961

**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/100959. The root cause is for node of `torch.ops.aten.max_pool2d_with_indices.default`, there are 2 output node as output tensor and max indices. So in its `node.meta["val"]` is a tuple of `FakeTensors` (For example: `'val': (FakeTensor(..., size=(1, 2, s1, s1)), FakeTensor(..., size=(1, 2, s1, s1), dtype=torch.int64))`). It will fail the check  of inserting observer since which only accept one `FakeTensor` case. 

**Test Plan**
```
python -m pytest test_quantize_pt2e.py -k test_max_pool2d_quantizer
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
Hi @jerryzh168, May I kindly ask a question about this piece of logic?https://github.com/pytorch/pytorch/blob/f08ddae8885bc703408b949642d4a5bee30efce8/torch/ao/quantization/_pt2e/prepare.py#L151-L155
Why we skip of inserting observer if the output is not a tensor? Does the change in this PR looks good to you?

**Comment by jerryzh168** [comment]
why do we change this? I don't think we want to insert observers for non-Tensor values

**Comment by jerryzh168** [comment]
oh I see, I think the skip logic is not correct, we should only skip inserting output observer in this case

File: torch/ao/quantization/_pt2e/prepare.py [hunk 1]
```diff
@@ -22,9 +22,14 @@ def _maybe_insert_input_and_output_observers_for_node(
...
     if "val" in node.meta:
+        def check_node_meta_val_fake_tensor(node_meta_val):
+            if isinstance(node_meta_val, (tuple, list)):
+                return all(check_node_meta_val_fake_tensor(element) for element in node_meta_val)
+            else:
+                return isinstance(node_meta_val, FakeTensor)
         output_is_a_tensor = (
             this_node_dtype_info is not None and
-            isinstance(node.meta["val"], FakeTensor)
+            check_node_meta_val_fake_tensor(node.meta["val"])
```
Comment (jerryzh168) [inline torch/ao/quantization/_pt2e/prepare.py:32]: not this one, L41 should be applied after L56

Comment (leslie-fang-intel) [reply torch/ao/quantization/_pt2e/prepare.py]: Looks like we can't simply move L41 after L56 due to the case https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/_pt2e/prepare.py#L30
When the node has no `target_dtype_info`, we should also skip the insert of input observer. Does my understanding right?

Comment (leslie-fang-intel) [reply torch/ao/quantization/_pt2e/prepare.py]: Move L41 after L56 and also add `skip_inserting_input_observers` to skip the insert of input observer when `node.meta["target_dtype_info"]` not set for this node. @jerryzh168 Please help to take a look of this PR again.

File: torch/ao/quantization/_pt2e/prepare.py [hunk 2]
```diff
@@ -29,11 +29,11 @@ def _maybe_insert_input_and_output_observers_for_node(
     else:
         output_is_a_tensor = this_node_dtype_info is not None
 
-    skip_inserting_observers = (
-        not output_is_a_tensor
+    skip_inserting_input_observers = (
```
Comment (jerryzh168) [inline torch/ao/quantization/_pt2e/prepare.py:32]: nit: this should probably be `skip_inserting_input_and_output_observers`

**Review by jerryzh168 — APPROVED** [review]
LGTM, we are landing a refactor diff right now though: https://github.com/pytorch/pytorch/pull/101041 maybe you can land after that is merged

**Comment by leslie-fang-intel** [comment]
It's for 

> LGTM, we are landing a refactor diff right now though: #101041 maybe you can land after that is merged

Sure, will land this PR after that.

**Review by jgong5 — APPROVED** [review]

File: torch/ao/quantization/_pt2e/prepare.py (hunk 2 above)
Comment (leslie-fang-intel) [reply torch/ao/quantization/_pt2e/prepare.py:158]: Changed and thanks.

**Pull request closed without merging**